React/Project structure.md

```diff
@@ -810,3 +816,164 @@ src
+   * Data fetching is now more flexible. You can fetch data directly in a Server Component (using standard `async/await`), or use RSC primitives like `fetch()`.
+   * `layout.tsx` can also fetch data and pass it to child components as props.
+7. API Routes in `app/api/`
+   * Replaces `pages/api/` for route handlers.
+   * Uses **Request** & **Response** objects instead of API handlers.
+8. `loading.tsx` and `error.tsx` for Better UX
+   * `loading.tsx` shows a skeleton loader while fetching data.
+   * `error.tsx` catches errors for that route.
+   * Works per-route or globally in `app/`.
+
+**Why it's better**
+
+1. More Flexible & Scalable
+   * Nested layouts and route groups let you organize complex UIs in a way that was harder or more verbose in the Pages Router.
+   * The file-system routing remains intuitive while providing new capabilities like colocation of components, hooks, and utilities.
+2. Performance & Bundle Size
+   * Server Components help keep initial payload sizes down by only shipping necessary code to the client.
+   * Automatic code splitting ensures only essential bits are loaded per route.
+3. Better Developer Experience
+   * Colocation of related files (tests, types, hooks) simplifies discovering and maintaining logic.
+   * The new data-fetching model (removing `getServerSideProps` and `getStaticProps`) feels more natural to React developers.
+4. Incremental Adoption
+   * You can run App Router and Pages Router in the same project if you are migrating gradually.
+   * This means no big-bang rewrite, so you can adopt new features at your own pace.
+
+### App Router structure
+
+Below is an example of a complete App Router folder structure, similar to what you have in the Pages Router section, but adapted to Next.js 13 best practices. This structure follows the **Colocation** principle, which encourages grouping related files together within their respective feature or route directories. This approach improves maintainability, reduces unnecessary imports, and keeps the project modular. You can learn more about Colocation in the [Next.js documentation](https://nextjs.org/docs/app/getting-started/project-structure#colocation). Note that this example includes route groups like `(authorization)`, `(admin)`, and `(marketing)`, a `[locale]` directory for **i18n**, plus an `api` folder for route handlers:
+
+```
+src
+├── app
+│   ├── api                              // Server API endpoints (App Router)
+│   │   ├── health                       // Example server endpoint: /api/health
+│   │   │   └── route.ts
+│   │   └── auth
+│   │       └── [..nextauth]
+│   │           └── route.ts            // NextAuth route
+│   └── [locale]                         // Top-level for i18n (e.g. /en, /fr)
+│       ├── layout.tsx                   // Root layout for everything under /[locale]
+│       ├── page.tsx                     // Root page -> /[locale]
+│       ├── _components                  // Shared "global" components
```

Comment:
Where would you put components that are used in the Root page, that should not be global? (home page components)